frontend/model/contracts/group.js

```diff
@@ -788,6 +850,67 @@ sbp('chelonia/defineContract', {
         })
       }
     },
+    'gi.contracts/group/addChatRoom': {
+      validate: objectMaybeOf({
+        chatRoomID: string,
+        name: string,
+        type: unionOf(...Object.values(chatRoomTypes).map(v => literalOf(v)))
+        // TODO: need to define 'editable' as boolean
+        // but 'boolean' TypeValidator doesn't work fine inside objectMaybeOf
+        // editable: boolean
+        // private: boolean
+      }),
+      process ({ data, meta }, { state, getters }) {
+        Vue.set(state.chatRooms, data.chatRoomID, {
+          creator: meta.identityContractID,
+          name: data.name,
+          type: data.type,
+          private: data.private,
+          editable: data.editable
+        })
+      }
+    },
+    'gi.contracts/group/removeChatRoom': {
+      validate: objectOf({
+        chatRoomID: string
+      }),
+      process ({ data, meta }, { state, getters }) {
+        Vue.delete(state.chatRooms, data.chatRoomID)
+      },
+      sideEffect ({ meta, data }, { state }) {
+        sbp('state/vuex/commit', 'removeContract', data.chatRoomID)
+      }
+    },
+    'gi.contracts/group/joinChatRoom': {
+      validate: objectOf({
+        chatRoomID: string
+      }),
+      process ({ data, meta }, { state, getters }) {},
+      async sideEffect ({ meta, data }, { state }) {
+        const rootState = sbp('state/vuex/state')
+        if (meta.username === rootState.loggedIn.username) {
+          sbp('okTurtles.data/set', 'JOINING_CHATROOM', true)
+          await sbp('state/enqueueContractSync', data.chatRoomID)
```

Comment:
Instead of `'state/enqueueContractSync'`, let's use `'gi.actions/contract/sync'`, which does the same thing, but we need to move the codebase over to using that, and eventually remove `'state/enqueueContractSync'`.

(And eventually this will also be renamed to something like `'chelonia/contract/sync'`)

Reply:
Got it.